### ISSUE TITLE
There seem to be some errors in the code

### ISSUE THREAD

**Issue by wushangbin** [XinyiYS/Robust-and-Fair-Federated-Learning#3]
1. In utils / utils.py, there seems to be a shrink error on lines 149-153.  I guess these codes should be in the loop of each batch. Is this right?
2. In RFFL_run.py, line 391,  Is it reasonable to use the validation set of each client to evaluate each model? Should a test set from server be used for evaluation?
3. In RFFL_run.py, line 291-293,  it seems that each iteration (a round in federated learning) starts from an initialization model. In this case, the training is ineffective. Is your code wrong?

**Comment by XinyiYS** [comment]
1. Yeah, you are right. Thanks. Fixing it. 
2. The `valid_loader` is in fact maintained by the server, _not_ from each client.
3. In lines 385-388, the models are updated accordingly.

**Comment by XinyiYS** [comment]
Closing due to inactivity. Feel free to open again if you have questions.
